### PR TITLE
Block Library: Do not display empty classic blocks if Classic block is not registered

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR updates the missing block to not render empty freeform blocks if the Classic block is not registered. 

This PR is part of the fix to address #52811.

## Why?
With this additional condition, if the TinyMCE removal experiment is turned on and the Classic block is intentionally not registered, we won't be displaying empty classic blocks.

## How?
We're moving a hook above to preserve the React [rule of hooks](https://legacy.reactjs.org/docs/hooks-rules.html), and adding an additional condition to render nothing if all conditions are true:

* the currently missing block has NO content
* the currently missing block is a `core/freeform` block
* the `core/freeform` block isn't registered - in the case when TinyMCE removal experiment is enabled

## Testing Instructions
* Follow the testing instructions of #52811.
* Insert a post with the following content:

```
<!-- wp:paragraph {"align":"center"} -->
<p class="has-text-align-center">Please be aware this is a test.</p>
<!-- /wp:paragraph -->

<!-- wp:freeform /-->

<!-- wp:paragraph {"align":"center"} -->
<p class="has-text-align-center">A second paragraph that has content inserted</p>
<!-- /wp:paragraph -->
```
* Verify that after refreshing the post page in the admin, you don't see this missing block warning: 

```
Your site doesn’t include support for the "core/freeform" block. You can leave this block intact or remove it entirely.
```

### Testing Instructions for Keyboard
None

## Screenshots or screencast 
None

### CODE REVIEW

File: packages/block-library/src/missing/edit.js [hunk 1]
```diff
@@ -64,6 +65,12 @@ function MissingBlockWarning( { attributes, convertToHTML, clientId } ) {
 			originalName
 		);
 		actions.push( convertToHtmlButton );
+	} else if (
+		! hasContent &&
+		! hasFreeformBlock &&
+		originalName === 'core/freeform'
+	) {
+		return null;
```
Comment (Mamaduka) [inline packages/block-library/src/missing/edit.js:73]: Can we use this check and return `null` early here? It's a bit hard to parse the message string assignment and return combo 😅

**Review by Mamaduka — COMMENTED** [review]
This tests well for me. Thanks, @tyxla!

The missing block is no longer displayed in the post content. But I see it in the List View.

Is this something that your other PRs will fix?

## Screenshot
CleanShot 2023-07-25 at 17 20 57

**Comment by tyxla** [comment]
Thanks for taking a look, @Mamaduka! 🙌 

> The missing block is no longer displayed in the post content. But I see it in the List View.

I think this is a sign that we might be fixing this in the wrong place. I'll keep this open but investigate a better solution.

In the meantime, the other PRs address other related issues and still deserve their own review if you have some spare time.

**Comment by Mamaduka** [comment]
> I think this is a sign that we might be fixing this in the wrong place. I'll keep this open but investigate a better solution.

That's what I was thinking as well. The example CPT template clearly doesn't define anything that could be interpreted as a freeform block.

> I think this is a sign that we might be fixing this in the wrong place. I'll keep this open but investigate a better solution.

Already on my to-do list, and I'll have a look tomorrow.

**Comment by tyxla** [comment]
Thanks @Mamaduka 🙌 

I'll be working to address this particular scenario in an alternative, more reliable way.

**Comment by tyxla** [comment]
@Mamaduka actually, I've given this a bit more thought and I think it makes sense to stay as-is. An empty freeform block shouldn't exist in the first place; if it's there for some reason, it only makes sense to be suggested to be removed in a transparent for the user way. The other PRs resolve the rest of the cases properly, so we should be good.

**Comment by Mamaduka** [comment]
> An empty freeform block shouldn't exist in the first place

I think that's the issue. When the experiment is disabled, there's no Freeform block since the CPT template just [defines two paragraphs](https://gitlab.com/cpl/test-case-52811/-/blob/main/register.php?ref_type=heads#L32-47). But when we enable the experiment, we see the "ghost" Freeform block.

**Comment by tyxla** [comment]
@Mamaduka Are you still seeing the issue with the PR to conditionally set the freeform handler merged? I tried to reproduce and can't anymore.

**Comment by Mamaduka** [comment]
I just re-tested the latest trunk and can no longer reproduce the issue.

**Comment by tyxla** [comment]
Awesome, thanks @Mamaduka. I'll wait for the original reporter of https://github.com/WordPress/gutenberg/issues/52811 to retest before considering the matter fixed and closing the issue.
